source/gravity_model/radial.cc

```diff
@@ -178,6 +183,159 @@ namespace aspect
+       * where G = gravitational constant, M_p = mass of planet, a_s = semimajor axis of satellite's orbit, b = angular rate of nonsynchronous rotation.
+       * r, theta and lambda are radial distance, polar angle and azimuthal angle, respectively.
+       * b [1/s] = b_NSR [m/yr] / (circumference of satellite [m]) / year_to_seconds [s/yr] * 2 * pi
+       *
+       * Notation of this potential equation is converted from spherical coordinates to cartesian coordinates.
+       * Therefore, gradient of potential is (3 G M_p) / (2 a_s^3) * ( 1 / 6 * ( x^2 + y^2 - 2 * z^2) + 1 / 2 * (C1*(x^2 + y^2) - 2 * C2 * x * y)))
+       * where C1 = cos(2*b*t) and C2 = sin(2*b*t)
+       */
+      const Tensor<1,dim> e_x = Point<dim>::unit_vector(0);
+      const Tensor<1,dim> e_y = Point<dim>::unit_vector(1);
+      const Tensor<1,dim> e_z = Point<dim>::unit_vector(2);
+
+      const double t = (this->simulator_is_past_initialization()) ? this->get_time() : 0.0;
+      const double x = p[0];
+      const double y = p[1];
+      const double z = p[2];
+
+      const double C1 = std::cos( 2. * b_NSR / R1 / year_in_seconds * t);
+      const double C2 = std::sin( 2. * b_NSR / R1 / year_in_seconds * t);
+
+      const double dTstar_over_dx = 1. / 6. * ( 2. * x );
+      const double dT0_over_dx    = 1. / 2. * ( 2. * C1 * x - 2. * C2 * y );
+
+      const double dTstar_over_dy = 1. / 6. * ( 2. * y );
+      const double dT0_over_dy    = 1. / 2. * ( -2. * C1 * y - 2. * C2 * x );
+
+      const double dTstar_over_dz = 1. / 6. * ( -4. * z );
+      const double dT0_over_dz    = 0;
+
+      const double G = aspect::constants::big_g;
+      const double T_factor = 3. * G * M_p / 2. / a_s / a_s / a_s;
+
+      const Tensor<1,dim> tidal_gravity = T_factor *
+                                          ( (dTstar_over_dx + dT0_over_dx) * e_x
+                                            + (dTstar_over_dy + dT0_over_dy) * e_y
+                                            + (dTstar_over_dz + dT0_over_dz) * e_z);
+
+      return RadialLinear<dim>::gravity_vector(p) + tidal_gravity;
+    }
+
+
+    template <int dim>
+    void
+    RadialLinearWithTidalPotential<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      RadialLinear<dim>::declare_parameters (prm);
+
+      prm.enter_subsection("Gravity model");
+      {
+        prm.enter_subsection("Radial linear with tidal potnetial");
+        {
+          prm.declare_entry ("Mass of planet", "1.898e27",
+                             Patterns::Double (),
+                             "Mass of satellte's host planet. "
+                             "Default value is for Europa, therefore, mass of Jupiter. "
+                             "Units is $kg$.");
+          prm.declare_entry ("Semimajor axis of satellite", "670900000",
+                             Patterns::Double (),
+                             "Length of semimajor axis of satellite. "
```

Comment:
Better to be more verbose:
 ```suggestion
                             "Length of the semimajor axis of the orbit of the modeled planet. "
```